src/core/plugin/luapi_application.h

```diff
@@ -56,6 +56,30 @@ extern "C" {
 #include <lualib.h>   // for luaL_openlibs
 }
 
+
+static std::tuple<std::optional<std::string>, std::vector<Element*>> getElementsFromHelper(Control* control,
+                                                                                           const std::string& type) {
+    std::vector<Element*> elements = {};
+    if (type == "layer") {
+        auto sel = control->getWindow()->getXournal()->getSelection();
+        if (sel) {
+            control->clearSelection();  // otherwise texts in the selection won't be recognized
+        }
+        elements = xoj::refElementContainer(control->getCurrentPage()->getSelectedLayer()->getElements());
+    } else if (type == "selection") {
+        auto sel = control->getWindow()->getXournal()->getSelection();
+        if (sel) {
+            elements = sel->getElements();
+        } else {
+            return std::make_tuple(std::make_optional("There is no selection"), elements);
+        }
+    } else {
+        return std::make_tuple(std::make_optional("Unknown argument for getting selection"), elements);
```

Comment:
You could add the `type` to the string here.

Reply:
But how would you build the string? (we're just passing the string so we don't have the `lua_...` formatting capabilities at that point)

Reply:
E.g. like this:
```c++
        std::stringstream msg;
        msg << "Unknown argument for getting selection: " << type;
        return std::make_tuple(std::make_optional(msg.str()), elements);
```

Reply:
:+1: Thanks.

(Somehow I tried to circumvent this dynamic string building at first, but I guess when dealing with user feedback we should do this.)